tool/tsh/app_cloud.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+// cloudApp is a dummy interface shared between awsApp, azureApp and gcpApp
+type cloudApp interface {
+	dummyCloudAppMethod()
+}
+
+// cloudAppImpl automatically implements cloudApp when inherited
+type cloudAppImpl struct {
+}
+
+func (_ cloudAppImpl) dummyCloudAppMethod() {
+}
+
+// cloudAppInfo contains common information for given cloudApp
+type cloudAppInfo struct {
+	// matchRouteToApp returns true if given tlsca.RouteToApp is recognized as a given cloud app kind.
+	matchRouteToApp func(tlsca.RouteToApp) bool
+	// newCloudApp creates a new instance of cloudApp
+	newCloudApp func(cf *CLIConf, profile *client.ProfileStatus, appRoute tlsca.RouteToApp) (cloudApp, error)
+	// cloudFriendlyName is a friendly name for given cloud
+	cloudFriendlyName string
+}
+
+func (info *cloudAppInfo) pickActiveCloudApp(cf *CLIConf) (cloudApp, error) {
```

Comment:
`pickActiveCloudApp` suggest that app is returns an active app but under the hood if app is not active we will fallback to login call.

Reply:
It is active once it is returned, but I do see your point.

Reply:
Renamed; I think we will be fine with a simpler names.